exporters/prometheus/exporter.go

```diff
@@ -87,7 +91,7 @@ type collector struct {
 	disableTargetInfo bool
 	targetInfo        prometheus.Metric
 	scopeInfos        map[instrumentation.Scope]prometheus.Metric
-	scopeInfoErrs     map[instrumentation.Scope]error
+	scopeInfosInvalid map[instrumentation.Scope]bool
```

Comment:
Can you change the `map[instrumentation.Scope]bool` to `map[instrumentation.Scope]struct{}`?
While I agree that this is idomatic, we are using `map[T]struct{}` everywhere else as it has lower memory footprint (and for consistency).